### DISCUSSION TITLE
Why does the Switch.Label contain an onClick property?

### DISCUSSION THREAD

**Discussion by imevanc** [tailwindlabs/headlessui#2542] [Ideas]
I noticed that the Switch.Label includes an onClick property, given that a label should be a p or a span tag, why have you added the [onClick](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/switch/switch.tsx#L69) here?

It can possibly create unwanted behaviours such as [this one](https://tailwindui.com/components/application-ui/forms/toggles#component-b99e1c97056ef7d294767c9def084f7b) that I can see on the tailwindui. When the user clicks the label part of the component (ie "Annual billing (Save 10%)"),  the toggle button is enabled/disabled. 

After a small research, I found out that behaviour is not there if the Switch.Label is encapsulated in another p or span tag, but I am not convinced that this is the best thing to do.

**Comment by imevanc** [comment]
Has anyone seen this? @adamwathan

**Comment by adamwathan** [comment]
Hey! This is how labels generally work in HTML — clicking the label activates or toggles the associated form control.

See this really minimal JSFiddle example:

https://jsfiddle.net/s8oL2f6a/

If you don't want that behavior for your switch, set the `passive` prop on the label:

https://headlessui.com/react/switch#using-a-custom-label

<img width="866" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/4323180/b475bb95-1cb7-40ec-ba05-2d35d0f21f2e">

**Comment by imevanc** [comment]
Cheers for clarifying it.
